atsign-foundation/noports
# refactor: move more code from ssh / npt handlers to handler_commons

**- What I did**

Please ignore branch name, the work in this PR needed to be done first, once this is merged, I will circle back to do the work originally intended for this branch.

- Refactored several shared steps between handle ssh and handle npt into handler commons
- Found some pthread related issues
  - Such as places where the shared worker atclient wasn't being locked
  - Handling of situations where a lock fails to lock on the main thread
  - The refresh devices iteration will be skipped if the lock fails

**- How I did it**

**- How to verify it**

- Tested on sshnp and npt
- Will run the daemon overnight to see if any crashes or if device entry is skipped too often.

**- Description for the changelog**
refactor: move more code from ssh / npt handlers to handler_commons
